.github/ci_workflows_docs.md

```diff
@@ -0,0 +1,110 @@
+# Ci scripts
+The documentaion below showcases each CI workflows' functionality.
+
+## .github/workflows/cuda.yml
+This script aims to test the project's CUDA support with the following different configurations:
+- Source build - Python 3.7, GCC compiler, CUDA 11.2.2
+- Pip install - Python 3.8, Clang compiler, CUDA 10.0
+- Wheel install - Python 3.9, GCC compiler, CUDA 9.0
+
+It triggers on pushing and creating pull requests to master.
+Set environment variables:
+  - github_actions: 'true'
+  - os_name: linux
+  - task: cuda
+  - conda_env: test-env 
+
+The steps are presented below
+### Setup or update software on host machine
+This step aims to setup the host machine with the necessary images and packages (Ubuntu image, Nvidia docker engine resources and docker)
+
+### Setup and run tests
+This step aims to run the tests after the setup is done. This is achieved by setting up all the necessary variables and then running the docker image with any of the configuration combinations specified above.
+
+## .github/workflows/linkchecker.yml
+As the name suggets, this step is meant to lint check. It is triggered with a cron schedule - once a day at 08:00am UTC. Otherwise, it can can be triggered manually.
```

Comment:
```suggestion
As the name suggets, this step is meant to link check. It is triggered with a cron schedule - once a day at 08:00am UTC. Otherwise, it can can be triggered manually.
```